### DISCUSSION TITLE
Use appsettings values as input or variable for a WF in designer

### DISCUSSION THREAD

**Discussion by dogcane** [elsa-workflows/elsa-core#5664] [Q&A]
Hi everybody, I'm using Elsa and WF designer (no "code" WF) to manage workflows and interactions between microservices. At the moment we've defined variables to specify the endpoints used in HttpRequest activities. Is it possible to read this values from the appsettings? is there a better way/approach to manage this case?

Thanks in advance!

**Comment by glime-ai** [comment]
Yes, it is possible to read values from the `appsettings.json` file in Elsa workflows, especially when using HttpRequest activities. You can achieve this by configuring your workflows to use variables that are populated from configuration settings.

Here's a general approach to manage this case:

1. Define your endpoint URLs in the `appsettings.json` file under a specific section.
2. In your application startup code (typically in `Program.cs` or `Startup.cs`), read these settings and add them as Elsa script variables or workflow variables.
3. In your workflow definitions, use these variables to configure the HttpRequest activities.

For example, your `appsettings.json` might look like this:

```json
{
  "Endpoints": {
    "ServiceA": "http://service-a/api",
    "ServiceB": "http://service-b/api"
  }
  // Other settings...
}
```

In your startup code, you would read these settings and add them as variables:

```csharp
public void ConfigureServices(IServiceCollection services)
{
    IConfiguration configuration = ...; // Obtain your configuration instance

    services.AddElsa(elsa =>
    {
        // Other Elsa configurations...

        // Add script variables from configuration
        elsa.AddScriptingOptions(options =>
        {
            var endpoints = configuration.GetSection("Endpoints").Get<Dictionary<string, string>>();
            foreach (var endpoint in endpoints)
            {
                options.ScriptEngine.GlobalVariables[endpoint.Key] = endpoint.Value;
            }
        });
    });
}
```

In your workflow designer, when setting up the HttpRequest activity, you would use expressions to reference these variables for the URL:

```
{{ Variables.ServiceA }}
```

This way, you can manage your endpoint URLs from a central configuration file, making it easier to maintain and update as needed.

Remember to ensure that your application has the necessary permissions to read from the `appsettings.json` file and that the configuration keys match those used in your workflow definitions.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Reply by dogcane** [reply]
Hi, I cannot find "AddScriptingOptions" method (I'm using Elsa 3.1.3).

Thanks.
